crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -952,17 +952,19 @@ where
         return Ok(true);
     }
 
-    let maybe_block_hash = client.hash(block_number_to_archive)?;
-
-    let Some(block_hash) = maybe_block_hash else {
-        warn!("Archiver detected an error: older block by number must always exist. ");
-
+    if block_number_to_archive > *best_archived_block_number + 1u32.into() {
+        // There is a gap between the best archived block and the block to archive.
+        // It's likely a fast sync.
         return Ok(false);
-    };
+    }
+
+    let block_hash = client
+        .hash(block_number_to_archive)?
+        .expect("Older block by number must always exist.");
 
     let block = client
         .block(block_hash)?
-        .expect("Older block by number must always exist");
+        .expect("Older block by hash must always exist.");
```

Comment:
Please no dots at the end of error message. This is not a paragraph of a larger text, we don't need to put dots at the end all the time.